Improve theming documentation (#1615)
vaadin/docs: articles/styling/custom-theme/styling-components.adoc

You might need to use more specific selectors to override the current theme's styles.
Use the browser inspector to identify the applied styles for the element that you wish to customize.

> See <<{articles}/styling/custom-theme/creating-custom-theme#, Creating a Custom Theme>> for instructions how to import the above styles into the local style scope of the Text Field component.

Reviewer: ```suggestion
See <<{articles}/styling/custom-theme/creating-custom-theme#, Creating a Custom Theme>> for instructions on how to import the above styles into the local style scope of the Text Field component.
```

}
----
+

> The `:host` selector also accepts a simple selector as an argument for it, scoping the selector to match the host only if the added selector matches it as well. It is useful, for example, for matching specific <<states>> and <<{articles}/styling/lumo/variants/component-variants#,Theme Variants>> of a component (using an attribute selector).

Reviewer: ```suggestion
The `:host` selector also accepts a simple selector as an argument, scoping the selector to match the host only if the added selector matches it as well. It is useful, for example, for matching specific <<states>> and <<{articles}/styling/lumo/variants/component-variants#,Theme Variants>> of a component (using an attribute selector).
```

One example of a such situation are Grid cells, which create a clipping stacking context.

All components that have an overlay sub-component have their own extension of the base Overlay component, for example “Dialog Overlay” (`<vaadin-dialog-overlay>`) and “Date Picker Overlay” (`<vaadin-date-picker-overlay>`).

> As an overlay is considered as a sub-component of the main component, <<{articles}/styling/lumo/variants/component-variants#, sub-component variants>> apply to them as well.

Reviewer: ```suggestion
As an overlay is considered to be a sub-component of the main component, <<{articles}/styling/lumo/variants/component-variants#, sub-component variants>> apply to them as well.
```

An input container provides a default slot for an HTML element that contains a value, as well as `prefix` and `suffix` named slots.
Depending on the main component, an element to display value is either `<input>` or `<textarea>`, except for Select which provides its own component.

> Same as <<overlays>>, an Input Container is considered as a sub-component of the main component, <<{articles}/styling/lumo/variants/component-variants#, sub-component variants>> apply to them as well.

Reviewer: ```suggestion
In the same way as <<overlays>>, an Input Container is considered to be a sub-component of the main component.
<<{articles}/styling/lumo/variants/component-variants#, Sub-component variants>> apply to them as well.
```

----
====

> You can also use the `theme` attribute (see <<{articles}/styling/custom-theme/custom-component-variants#,custom component variants>>) as a scoping selector for your style overrides, as shown in the example below.

Reviewer: ```suggestion
You can also use the `theme` attribute (see <<{articles}/styling/custom-theme/custom-component-variants#,custom component variants>>) as a scoping selector for your style overrides, as shown in the example that follows.
```
